packages/gatsby/src/utils/parcel/compile-gatsby-files.ts

```diff
@@ -38,16 +38,48 @@ export function constructParcel(siteRoot: string): Parcel {
  * and output in `<SITE_ROOT>/.cache/compiled`.
  */
 export async function compileGatsbyFiles(siteRoot: string): Promise<void> {
-  const parcel = constructParcel(siteRoot)
-
   try {
-    await ensureDir(`${siteRoot}/${COMPILED_CACHE_DIR}`)
+    const parcel = constructParcel(siteRoot)
+    const distDir = `${siteRoot}/${COMPILED_CACHE_DIR}`
+    await ensureDir(distDir)
+    await emptyDir(distDir)
```

Comment:
Of importance - added `emptyDir` to clean out potentially stale files in compiled dir from previous compilations